Python/pystate.c

```diff
@@ -324,6 +326,7 @@ interpreter_clear(PyInterpreterState *interp, PyThreadState *tstate)
 
     /* Last garbage collection on this interpreter */
     _PyGC_CollectNoFail(tstate);
+    PyThreadState_Clear(tstate);
```

Comment:
For safety, I would prefer to ensure that the tstate belongs to the interpreter if PyInterpreterState_Clear() is called from a different interpreter (even if I don't know if it's currently possible):

```suggestion
    // Don't clear tstate if it belongs to another interpreter
    if (tstate->interp == interp) {
        PyThreadState_Clear(tstate);
    }
```